src/main/java/io/kestra/plugin/gcp/cli/Commands.java

```diff
@@ -0,0 +1,75 @@
+package io.kestra.plugin.gcp.cli;
+
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
+import io.kestra.plugin.scripts.exec.scripts.models.RunnerType;
+import io.kestra.plugin.scripts.exec.scripts.models.ScriptOutput;
+import io.kestra.plugin.scripts.exec.scripts.runners.CommandsWrapper;
+import io.kestra.plugin.scripts.exec.scripts.services.ScriptService;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "Execute gcp commands."
+)
+public class Commands extends Task implements RunnableTask<ScriptOutput> {
+    @PluginProperty
+    protected String serviceAccount;
+
+    @Schema(
+            title = "The commands to run"
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @NotEmpty
+    protected List<String> commands;
+    @Schema(
+            title = "Additional environment variables for the current process."
+    )
+    @PluginProperty(
+            additionalProperties = String.class,
+            dynamic = true
+    )
+    protected Map<String, String> env;
+
+    @Override
+    public ScriptOutput run(RunContext runContext) throws Exception {
+        CommandsWrapper commands = new CommandsWrapper(runContext).withDockerOptions(
+                DockerOptions.builder()
+                .image("google/cloud-sdk")
```

Comment:
You should also expose docker option as a property as some may want to customize them

Reply:
Why would you want to customize docker options ? Also I first went for a script-based implementation (extending AbstractExec...) but @tchiotludo told me not to to prevent bringing useless properties in the task

Reply:
the dockeroptions can make sense I think, we can bring them (using a custom registry for example)

Reply:
:+1: